tests/test_log.py

```diff
@@ -72,3 +74,273 @@ def test_interceptor(self):
         _LOG.info('blabla')
         logger.shutdown()
         self.assertEqual(logger.records_lost, 0)
+
+    def test_json_list_format(self):
+        config = {
+            'log': {
+                'type': 'json',
+                'format': [
+                    {
+                        'asctime': '@timestamp',
+                    },
+                    {
+                        'levelname': 'level'
+                    },
+                    'message'
+                ],
+                'static_fields': {
+                    'app': 'patroni'
+                }
+            }
+        }
+
+        test_message = 'test json logging in case of list format'
+
+        stderr_output = StringIO()
+        original_stderr = sys.stderr
+        sys.stderr = stderr_output
+        sys.argv = ['patroni.py']
+        os.environ[Config.PATRONI_CONFIG_VARIABLE] = yaml.dump(config, default_flow_style=False)
+
+        logger = PatroniLogger()
+        patroni_config = Config(None)
+        logger.reload_config(patroni_config['log'])
+
+        logger.start()
+        _LOG.info(test_message)
+        sys.stderr = original_stderr
+        logger.shutdown()
```

Comment:
I was talking about using `patch('sys.stderr', ...)`:
```suggestion
        config = {
            'type': 'json',
            'format': [
                {'asctime': '@timestamp'},
                {'levelname': 'level'},
                'message'
            ],
            'static_fields': {
                'app': 'patroni'
            }
        }

        test_message = 'test json logging in case of list format'

        stderr_output = StringIO()
        with patch('sys.stderr', stderr_output):
            logger = PatroniLogger()
            logger.reload_config(config)
            _LOG.info(test_message)
```